### ISSUE TITLE
CaptureController::doSessionTokenAction should forward request data

### ISSUE THREAD

**Issue by Tetragramat** [Payum/PayumBundle#427]
If user is redirected from payment gateway with post or query data, then all values are discarded with redirect to the CaptureController::doAction.

This would be usefull because some payment gateways does not support dynamic return url.

**Comment by makasim** [comment]
I'd say the gateway have to store all data required (including one from request's query) before redirecting to doAction

**Comment by Tetragramat** [comment]
I agree, it is better than forwarding those data. Where it should be stored? In session or in payment model? Gateway is not loaded in doSessionTokenAction.

**Comment by makasim** [comment]
It is supposed to be stored in payment model and persited by storage (with a help of storage extension).

**Comment by Tetragramat** [comment]
So does that mean that there in doSessionTokenAction should be executed some request on gateway which will store values in payment model and store extension will persist it?

**Comment by makasim** [comment]
Sorry I thought the controller called a payum gateway. But it is not. 

The request has to be proxied with all possible data indeed. and I think if the original request is not GET we have to do a post redirect.

**Comment by Tetragramat** [comment]
By "payment gateways" in OP i meant something like Paypal/Stripe/Klarna. In this case is it something like Payeeze/First Data - FIO Banka scheme (it is unclear what is it's real name).

So, when it is clear now. Will be this feature in PayumBundle? I'm not sure If I will be able to efectively do proxy request.

**Comment by makasim** [comment]
I don't have time to implement it myself though maybe some one else pick this up and do a PR>
